### PR TITLE
stm32/boards: Add VCC_GND_F407ZG board.

### PR DESCRIPTION
**Pull request by mcauser**
Based on: https://github.com/mcauser/VCC_GND_F407ZG

### CODE REVIEW

**Comment by mcauser** [comment]
Rebased and tested against MPY v1.15

```
MicroPython v1.15-118-g0c36be284 on 2021-05-15; VCC-GND STM32F407ZG with STM32F407ZG
Type "help()" for more information.
```

**Comment by mcauser** [comment]
There's 2x VCC-GND Studio board PRs, VET6 + ZET6. This PR is for the ZET6 - the bigger one.

ZET6

File: ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,171 @@
+#define MICROPY_HW_BOARD_NAME       "VCC-GND STM32F407ZG"
+#define MICROPY_HW_MCU_NAME         "STM32F407ZG"
+#define MICROPY_HW_FLASH_FS_LABEL   "VCCGNDF407ZG"
+
+// 1 = use internal flash (1 MByte)
+// 0 = use onboard SPI flash (512 KByte) Winbond W25X40
+#define MICROPY_HW_ENABLE_INTERNAL_FLASH_STORAGE (1)
```
Comment (dpgeorge) [inline ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h:7]: why not enable SPI flash by default?

**Review by dpgeorge — COMMENTED** [review]
See also comments on #5531 about "buses" and alignment of macros.

File: ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h (hunk 1 above)
Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h:7]: The onboard VG flash is 1MB and the ext SPI flash is only 512KB.
... I picked the bigger one as the default.

Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h:7]: It'd be nice if they could both be enabled/used, one mounted as /flash and the other /spiflash _or something like that._
Like we can with /flash and /sd.
Only one would be mounted as the PYBOARD volume.

Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h:7]: I noticed a few occurrences of `bdev.c` in various boards. Perhaps could be moved up out of boards and made generic, configured on the board level with presence of new macros. Standard SPI, QSPI, soft and hard interfaces.

**Comment by mcauser** [comment]
- [x] Add vendor urls
- [x] Align macros
- [x] Busses -> buses

File: ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h (hunk 1 above)
Comment (dpgeorge) [reply ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h:12]: The available internal storage won't be 1Mbyte, right?  It'll be something small like 256kbyte.  I think it makes sense to enable external by default here.

> I noticed a few occurrences of `bdev.c` in various boards. Perhaps could be moved up out of boards and made generic, configured on the board level with presence of new macros. Standard SPI, QSPI, soft and hard interfaces

Yes, maybe... we can do that separately.

Comment (mcauser) [reply ports/stm32/boards/VCC_GND_F407ZG/mpconfigboard.h]: Enabled SPI flash as the default.

**Comment by dpgeorge** [comment]
Nice and clean commit with correct commit message, thank you!  Merged in 1ca66efbf7be4f57ab31df5ecb81d27da4aefd0d

**Comment by mcauser** [comment]
Thats how I roll 🎲 🎲

**Pull request closed without merging**